src/core/proj/qgscoordinatetransform.cpp

```diff
@@ -538,16 +540,28 @@ QgsRectangle QgsCoordinateTransform::transformBoundingBox( const QgsRectangle &r
         throw QgsCsException( QObject::tr( "Could not transform bounding box to target CRS" ) );
       yMax = 90 - EPS;
     }
+    if (xMin < -180)
+    {
+      if (xMax < -180)
+        throw QgsCsException(QObject::tr("Could not transform bounding box to target CRS"));
```

Comment:
The exception could be more talkative by letting know the error and pinpointing it. But the same comment could go for any of the exceptions around this code.